dags/fx_usage_report.py

```diff
@@ -17,24 +22,51 @@
 
 dag = DAG('fx_usage_report', default_args=default_args, schedule_interval='@weekly')
 
+
 wait_for_main_summary = ExternalTaskSensor(
     task_id='wait_for_main_summary',
     external_dag_id='main_summary',
     external_task_id='main_summary',
     execution_delta=timedelta(days=-7, hours=-1), # main_summary waits one hour, execution date is beginning of the week
     dag=dag)
 
-usage_report = EMRSparkOperator(
+
+cluster_name = 'fx-usage-report-dataproc-cluster'
+gcp_conn_id = 'google_cloud_airflow_dataproc'
+
+# AWS credentials to read/write from output bucket
+aws_conn_id = 'aws_prod_fx_usage_report'
+aws_access_key, aws_secret_key, session = AwsHook(aws_conn_id).get_credentials()
+
+output_bucket = 'net-mozaws-prod-us-west-2-data-public'
+
+usage_report = SubDagOperator(
     task_id="fx_usage_report",
-    job_name="Fx Usage Report",
-    execution_timeout=timedelta(hours=4),
-    instance_count=10,
-    owner="frank@mozilla.com",
-    email=["telemetry-alerts@mozilla.com", "frank@mozilla.com", "shong@mozilla.com"],
-    env={"date": DS_WEEKLY,
-         "bucket": "net-mozaws-prod-us-west-2-data-public",
-         "deploy_environment": "{{ task.__class__.deploy_environment }}"},
-    uri="https://raw.githubusercontent.com/mozilla/telemetry-airflow/master/jobs/fx_usage_report.sh",
-    dag=dag)
+    dag=dag,
+    subdag = moz_dataproc_scriptrunner(
+        parent_dag_name=dag.dag_id,
+        dag_name='fx_usage_report',
+        default_args=default_args,
+        cluster_name=cluster_name,
+        service_account='dataproc-runner-prod@airflow-dataproc.iam.gserviceaccount.com',
+        job_name="Fx_Usage_Report",
+        uri="https://raw.githubusercontent.com/mozilla/telemetry-airflow/master/jobs/fx_usage_report.sh",
+        env={"date": DS_WEEKLY,
+             "bucket": output_bucket,
+             "PYTHONPATH": "/usr/lib/spark/python/lib/pyspark.zip",
+             "deploy_environment": "{{ task.__class__.deploy_environment }}",
+             # These env variables are needed in addition to the s3a configs, since some code uses boto to list bucket objects
+             "AWS_ACCESS_KEY_ID": aws_access_key,
+             "AWS_SECRET_ACCESS_KEY": aws_secret_key
+        },
+        gcp_conn_id=gcp_conn_id,
+        # This should is used to set the s3a configs for read/write to s3 for non boto calls
+        aws_conn_id=aws_conn_id,
+        num_workers=9,
+        worker_machine_type='n1-standard-16',
+        image_version='1.3',
+        init_actions_uris=['gs://moz-fx-data-prod-airflow-dataproc-artifacts/bootstrap/fx_usage_init.sh'],
```

Comment:
This seems to be a bit of a leaky interface i.e. we're already exposing `artifact_bucket` as a config parameter and setting the default bootstrap to a path within that. Maybe it would be better to have this config option be e.g. `bootstrap_actions` and by convention make these values object paths within the artifact bucket (perhaps also prefixed by `bootstrap/`). As-is this is of course more general, and appears to map to the upstream version.